### PR TITLE
fix(sinks): Sinks webhdfs is not built

### PR DESCRIPTION
**Pull request by Xuanwo**
This pull request addresses the issue of webhdfs sinks not being built.

Additionally, it updates OpenDAL to its latest version. For more information on the latest version, please refer to the following changelog: https://docs.rs/opendal/0.30.2/opendal/docs/changelog/index.html.

Close: https://github.com/vectordotdev/vector/pull/16779

### CODE REVIEW

**Comment by Xuanwo** [comment]
Hello, @spencergilbert. The `check components docs` failed in the CI process but I am unable to reproduce it locally. Do you have any suggestions or ideas on how to resolve this issue?

```shell
cargo build --quiet
target/debug/vector generate-schema > /tmp/vector-config-schema.json 2>/dev/null
cargo vdev generate component-docs /tmp/vector-config-schema.json \
	>/dev/null
/home/runner/work/vector/vector/scripts/generate-component-docs.rb:368:in `get_schema_metadata': undefined method `dig' for "logical_name":String (NoMethodError)
	from /home/runner/work/vector/vector/scripts/generate-component-docs.rb:5[21](https://github.com/vectordotdev/vector/actions/runs/4413322355/jobs/7733664095#step:15:22):in `expand_schema_references'
	from /home/runner/work/vector/vector/scripts/generate-component-docs.rb:572:in `block (2 levels) in expand_schema_references'
	from /home/runner/work/vector/vector/scripts/generate-component-docs.rb:571:in `transform_values'
	from /home/runner/work/vector/vector/scripts/generate-component-docs.rb:571:in `block in expand_schema_references'
	from /home/runner/work/vector/vector/scripts/generate-component-docs.rb:536:in `loop'
	from /home/runner/work/vector/vector/scripts/generate-component-docs.rb:536:in `expand_schema_references'
	from /home/runner/work/vector/vector/scripts/generate-component-docs.rb:584:in `block (2 levels) in expand_schema_references'
	from /home/runner/work/vector/vector/scripts/generate-component-docs.rb:583:in `map'
	from /home/runner/work/vector/vector/scripts/generate-component-docs.rb:583:in `block in expand_schema_references'
	from /home/runner/work/vector/vector/scripts/generate-component-docs.rb:536:in `loop'
	from /home/runner/work/vector/vector/scripts/generate-component-docs.rb:536:in `expand_schema_references'
	from /home/runner/work/vector/vector/scripts/generate-component-docs.rb:584:in `block (2 levels) in expand_schema_references'
	from /home/runner/work/vector/vector/scripts/generate-component-docs.rb:583:in `map'
	from /home/runner/work/vector/vector/scripts/generate-component-docs.rb:583:in `block in expand_schema_references'
	from /home/runner/work/vector/vector/scripts/generate-component-docs.rb:536:in `loop'
	from /home/runner/work/vector/vector/scripts/generate-component-docs.rb:536:in `expand_schema_references'
	from /home/runner/work/vector/vector/scripts/generate-component-docs.rb:584:in `block (2 levels) in expand_schema_references'
	from /home/runner/work/vector/vector/scripts/generate-component-docs.rb:583:in `map'
	from /home/runner/work/vector/vector/scripts/generate-component-docs.rb:583:in `block in expand_schema_references'
	from /home/runner/work/vector/vector/scripts/generate-component-docs.rb:536:in `loop'
	from /home/runner/work/vector/vector/scripts/generate-component-docs.rb:536:in `expand_schema_references'
	from /home/runner/work/vector/vector/scripts/generate-component-docs.rb:595:in `block (2 levels) in expand_schema_references'
	from /home/runner/work/vector/vector/scripts/generate-component-docs.rb:594:in `map'
	from /home/runner/work/vector/vector/scripts/generate-component-docs.rb:594:in `block in expand_schema_references'
	from /home/runner/work/vector/vector/scripts/generate-component-docs.rb:536:in `loop'
	from /home/runner/work/vector/vector/scripts/generate-component-docs.rb:536:in `expand_schema_references'
	from /home/runner/work/vector/vector/scripts/generate-component-docs.rb:584:in `block (2 levels) in expand_schema_references'
	from /home/runner/work/vector/vector/scripts/generate-component-docs.rb:583:in `map'
	from /home/runner/work/vector/vector/scripts/generate-component-docs.rb:583:in `block in expand_schema_references'
	from /home/runner/work/vector/vector/scripts/generate-component-docs.rb:536:in `loop'
	from /home/runner/work/vector/vector/scripts/generate-component-docs.rb:536:in `expand_schema_references'
	from /home/runner/work/vector/vector/scripts/generate-component-docs.rb:740:in `resolve_schema'
	from /home/runner/work/vector/vector/scripts/generate-component-docs.rb:724:in `resolve_schema_by_name'
	from /home/runner/work/vector/vector/scripts/generate-component-docs.rb:1679:in `render_and_import_schema'
	from /home/runner/work/vector/vector/scripts/generate-component-docs.rb:17[23](https://github.com/vectordotdev/vector/actions/runs/4413322355/jobs/7733664095#step:15:24):in `render_and_import_base_component_schema'
	from /home/runner/work/vector/vector/scripts/generate-component-docs.rb:1771:in `block in <main>'
	from /home/runner/work/vector/vector/scripts/generate-component-docs.rb:1770:in `each'
	from /home/runner/work/vector/vector/scripts/generate-component-docs.rb:1770:in `<main>'
Error: command: "/home/runner/work/vector/vector/scripts/generate-component-docs.rb" "/tmp/vector-config-schema.json"
  failed with exit code: 1
```

File: Cargo.toml [hunk 1]
```diff
@@ -643,6 +643,7 @@ sinks-logs = [
   "sinks-socket",
   "sinks-splunk_hec",
   "sinks-vector",
+  "sinks-webhdfs",
```
Comment (jszwedko) [inline Cargo.toml:646]: Doh, thanks @Xuanwo .

**Review by jszwedko — COMMENTED** [review]
Thanks for fixing that and updating the library @Xuanwo !

**Comment by jszwedko** [comment]
> Hello, @spencergilbert. The check components docs failed in the CI process but I am unable to reproduce it locally. Do you have any suggestions or ideas on how to resolve this issue?

This is a sporadic failure that we've been investigating. We also haven't been able to reproduce locally. I restarted it.

**Pull request merged**